File: trunk/crescent/src/engine/static.lua
require 'i18n'

--
-- Goods
-- 
goods = {}
local function good(name, can_buy, initial_prices, raw_material)
   local g = {
      name = name,
      can_buy = can_buy,
      initial_prices = initial_prices,
      raw_material = raw_material,
   }
   table.insert(goods, g)
   return g
end

food =      good(_'Food',      true, {  1, 8 })
cotton =    good(_'Cotton',    true, {  4, 6 })
barley =    good(_'Barley',    true, {  4, 6 })
olives =    good(_'Olives',    true, {  6, 8 })
wood =      good(_'Wood',      true, {  1, 6 })
copper =    good(_'Copper',    true, {  5, 8 })
mud =       good(_'Mud',       true, {  1, 5 })
tin =       good(_'Tin',       true, {  7, 9 })
gold =      good(_'Gold',      true, { 15,16 })
clothes =   good(_'Clothes',   true, { 12,13 }, { cotton })
beer =      good(_'Beer',      true, { 10,11 }, { beer })
olive_oil = good(_'Olive Oil', true, { 12,13 }, { olives })
furniture = good(_'Furniture', true, { 10,11 }, { wood })
utensils =  good(_'Utensils',  true, { 12,13 }, { copper })
bricks =    good(_'Bricks',    true, {  1, 8 }, { mud })
riding =    good(_'Animals',   true, {  5, 8 })
bronze =    good(_'Bronze',    true, {  9,10 }, { copper, tin })
weapons =   good(_'Weapons',   true, {  6, 7 }, { wood, bronze })
scrolls =   good(_'Scrolls',   false)
prayers =   good(_'Prayers',   false)

-- 
-- Terrains
--
terrains = {}
local function terrain(name, code, ovl, cost_to_enter, defensive_bonus, production, switch_status)
   local function good_production(p)
      if not p then return {} end
      local pp = {}
      for i,v in ipairs(p) do pp[goods[i]] = v end
      return pp
   end
   local t = {
      name = name,
		ovl = ovl,
      code = code,
      production = good_production(production),
      switch_status = switch_status,
   }
   if switch_status then switch_status.switch_status = t end -- swap
   table.insert(terrains, t)
   return t
end

--                 Name               Code  MoveCost      Barley      Mud
--                                        Ovl*  Defence       Olives      Tin
--                                                  Food        Wood        Gold
--                                                     Cotton      Copper
tundra   = terrain(_'Tundra',         1, 12, 1, 0, { 2, 0, 0, 0, 0, 2, 2, 1, 0 })
desert   = terrain(_'Desert',         2, 13, 1, 0, { 1, 0, 0, 2, 0, 1, 3, 2, 0 })
plains   = terrain(_'Planis',         3, 14, 1, 0, { 4, 2, 1, 0, 0, 1, 1, 0, 0 })
prairie  = terrain(_'Prairie',        4, 15, 1, 0, { 3, 3, 0, 0, 0, 0, 0, 0, 0 })
steppe   = terrain(_'Steppe',         5, 16, 1, 0, { 2, 2, 3, 0, 0, 0, 0, 0, 0 })
marsh    = terrain(_'Marsh',          6, 17, 2, 1, { 3, 1, 2, 0, 0, 2, 4, 2, 0 })

boreal_f = terrain(_'Boreal forest',  7,  6, 2, 2, { 1, 0, 0, 0, 2, 1, 1, 1, 0 }, tundra)
scrub    = terrain(_'Scrubland',      8,  7, 1, 2, { 1, 0, 0, 4, 1, 1, 2, 1, 0 }, desert)
mixed_f  = terrain(_'Mixed forest',   9,  8, 2, 2, { 2, 1, 0, 0, 3, 0, 1, 0, 0 }, plains)
savannah = terrain(_'Savannah',      10,  9, 1, 1, { 1, 1, 0, 0, 2, 0, 1, 0, 0 }, prairie)
woodland = terrain(_'Woodland',      11, 10, 2, 2, { 1, 0, 1, 0, 2, 0, 1, 0, 0 }, steppe)
swamp    = terrain(_'Swamp',         12, 11, 3, 3, { 2, 0, 0, 0, 1, 1, 3, 2, 0 }, marsh)

arctic   = terrain(_'Arctic',        13,  5, 2, 0, { 0, 0, 0, 0, 0, 1, 0, 1, 1 })
sea      = terrain(_'Sea',           14,  2, 1, 0, { 3, 0, 0, 0, 0, 0, 0, 0, 0 })
ocean    = terrain(_'Ocean',         15,  1, 1, 0, { 3, 0, 0, 0, 0, 0, 0, 0, 0 })
hills    = terrain(_'Hills',         16,  4, 2, 4, { 1, 0, 0, 0, 0, 3, 0, 3, 0 })
mountain = terrain(_'Mountains',     17,  3, 3, 6, { 0, 0, 0, 0, 0, 3, 0, 3, 1 })
-- *Ovl: Overlap for terrains.


--
-- Military classes
--
militaries = {}
local function military(name, ship, moves, attack, defense, cargo)
local m = {
   name = name,
   ship = ship,
   moves = moves,
   attack = attack,
   defense = defense,
   cargo = cargo,
}
table.insert(militaries, m)
return m
end

peasant         = military(_'Peasant',                 false, 1, 0, 1, 0)
colonist        = military(_'Colonist',                false, 1, 0, 1, 0)
warrior         = military(_'Warrior',                 false, 1, 2, 2, 0)
nomad           = military(_'Nomad',                   false, 4, 1, 1, 0)
chariot         = military(_'Chariot',                 false, 4, 3, 3, 0)
pikeman         = military(_'Pikeman',                 false, 1, 3, 3, 0)
mtd_pikeman     = military(_'Mounted Pikeman',         false, 4, 4, 4, 0)
armored_warrior = military(_'Armed Warrior',           false, 1, 4, 4, 0)
cataphract      = military(_'Cataphract',              false, 4, 5, 5, 0)
catapult        = military(_'Catapult',                false, 1, 7, 5, 0)

clan_member     = military(_'Clan Member',             false, 1, 1, 1, 0)
clan_armed      = military(_'Armed Clan Member',       false, 1, 2, 2, 0)
clan_mounted    = military(_'Clan Member Rider',       false, 4, 2, 2, 0)
clan_mtd_armed  = military(_'Armed Clan Member Armed', false, 4, 3, 3, 0)

elite_soldier   = military(_'Elite Soldier',           false, 1, 4, 4, 0)
elite_chariot   = military(_'Elite Chariot',           false, 4, 5, 5, 0)
faraoh_soldier  = military(_"Faraoh's Guard",          false, 1, 5, 5, 0)
faraoh_chariot  = military(_"Faraoh's Chariot Guard",  false, 4, 6, 6, 0)

penteconter     = military(_'Penteconter',             true,  4, 0, 2, 2)
merchantman     = military(_'Merchantman',             true,  6, 0, 6, 5)
monoreme        = military(_'Monoreme',                true,  4, 1, 3, 0)
bireme          = military(_'Bireme',                  true,  5, 3, 5, 0)
trireme         = military(_'Trireme',                 true,  6, 4, 6, 0)
galley          = military(_'Galley',                  true,  5, 7, 8, 6)
pirate_ship     = military(_'Pirate Ship',             true,  8, 4, 8, 2)


-- 
-- Jobs
--
local function job(name)
   return {
      name = name,
   }
end

farmer       = job(_'Farmer')
fisherman    = job(_'Fisherman')
cotton_pl    = job(_'Cotton Planter')
barley_pl    = job(_'Barley Planter')
olive_pl     = job(_'Olives Planter')
lumberjack   = job(_'Lumberjack')
copper_miner = job(_'Copper Miner')
mudder       = job(_'Mudder')
tin_miner    = job(_'Tin Miner')
prospector   = job(_'Prospector')
weaver       = job(_'Weaver')
brewer       = job(_'Brewer')
oil_presser  = job(_'Oil Presser')
carpenter    = job(_'Carpenter')
coppersmith  = job(_'Coppersmith')
potter       = job(_'Potter')
metallurgic  = job(_'Metallurgic')
weaponsmith  = job(_'Weaponsmith')
scribe       = job(_'Scribe')
priest       = job(_'Priest')

--
-- Buildings
--
local function building(name, job, raw_good, good, multiplier, cost, 
      copper, min_colony)
   local b = {
      name = name,
      job = job,
      raw_good = raw_good,
      good = good,
      multiplier = mutiplier or 1, -- 3 = factory double production
      cost = cost or 0, -- 0 = starting building
      copper = copper or 0,
      min_colony = min_colony or 0,
   }
   if job then 
      job.building = b
      job.raw_good = raw_good
      job.good = good
   end
   return b
end

weaver_1    = building(_"Weaver's House", weaver, cotton, clothes)
weaver_2    = building(_"Weaver's Shop", weaver, cotton, clothes, 2, 64, 20)
weaver_3    = building(_"Cloth Mill", weaver, cotton, clothes, 3, 160, 100, 8)
brewer_1    = building(_"Brewer's House", brewer, barley, beer)
brewer_2    = building(_"Brewer's Shop", brewer, barley, beer, 2, 64, 20)
brewer_3    = building(_"Beer Mill", brewer, barley, beer, 3, 160, 100, 8)
oil_1       = building(_"Small Oil Press", oil_presser, olives, olive_oil)
oil_2       = building(_"Big Oil Press", oil_presser, olives, olive_oil, 2, 64, 20)
oil_3       = building(_"Olive Oil Mill", oil_presser, olives, olive_oil, 3, 160, 100, 8)
carpenter_1 = building(_"Carpenter's House", carpenter, wood, furniture)
carpenter_2 = building(_"Carpenter's Shop", carpenter, wood, furniture, 2, 64, 20)
carpenter_3 = building(_"Lumber Mill", carpenter, wood, furniture, 3, 160, 100, 8)
copper_1    = building(_"Coppersmith's House", coppersmith, copper, utensils)
copper_2    = building(_"Coppersmith's Shop", coppersmith, copper, utensils, 2, 64, 20)
copper_3    = building(_"Utensils Factory", coppersmith, copper, utensils, 3, 160, 100, 8)
brick_1     = building(_"Small Klin", potter, mud, bricks)
brick_2     = building(_"Big Klin", potter, mud, bricks, 2, 64, 20)
brick_3     = building(_"Brick Factory", potter, mud, bricks, 3, 160, 100, 8)
stable      = building(_"Stable", nil, food, horses, 2, 64, 20)
bronze_1    = building(_"Metallurgic's Shop", metallurgic, { copper, tin }, bronze)
bronze_2    = building(_"Metalworks", metallurgic, { copper, tin }, bronze, 2, 64, 20)
bronze_3    = building(_"Machine Shop", metallurgic, { copper, tin }, bronze, 3, 160, 100, 8)
scribe_1    = building(_"Tablet House", scribe, nil, scrolls)
scribe_2    = building(_"Gymnasium", scribe, nil, scrolls, 2, 64, 20)
scribe_3    = building(_"Academy", scribe, nil, scrolls, 3, 160, 100, 8)
temple_1    = building(_"Worship Yard", priest, nil, prayers, 1, 80)
temple_2    = building(_"Shrine", priest, nil, prayers, 2, 64, 20)
temple_3    = building(_"Temple", priest, nil, prayers, 3, 160, 100, 8)
embassy     = building(_"Embassy", nil, nil, nil, 3, 160, 100, 8)
docks_1     = building(_"Docks", nil, nil, nil, 1, 52)
docks_2     = building(_"Drydock", nil, nil, nil, 1, 80, 50, 4)
docks_3     = building(_"Shipyard", nil, nil, nil, 1, 240, 100, 8)
wall_1      = building(_"Wooden Wall", nil, nil, nil, 1, 64)
wall_2      = building(_"Stone Wall", nil, nil, nil, 2, 120, 100)
wall_3      = building(_"Masonry Wall", nil, nil, nil, 3, 240, 200)
library     = building(_"Library", nil, nil, nil, 0, 120, 20)
oracle      = building(_"Oracle", nil, nil, nil, 0, 120, 20)
warehouse_1 = building(_"Warehouse", nil, nil, nil, 2, 80)
warehouse_2 = building(_"Big Warehouse", nil, nil, nil, 3, 120, 20)
palace      = building(_"Palace", nil, nil, nil, 0, 240, 200)
monument    = building(_"Monument", nil, nil, nil, 0, 320, 300)

File: fccaruby/fcca/static.rb
require 'i18n'

#
# Goods
# 
class Good

  @@all = []

  attr_reader :name, :can_buy, :initial_prices, :raw_material, :all

  def initialize(name, can_buy, initial_prices=nil, raw_material=nil)
    @name, @can_buy, @initial_prices, @raw_material = name, can_buy, initial_prices, raw_material
    @@all << self
  end

  def Good.all; @@all; end

end

Food =      Good.new(_('Food'),      true, [  1, 8 ])
Cotton =    Good.new(_('Cotton'),    true, [  4, 6 ])
Barley =    Good.new(_('Barley'),    true, [  4, 6 ])
Olives =    Good.new(_('Olives'),    true, [  6, 8 ])
Wood =      Good.new(_('Wood'),      true, [  1, 6 ])
Copper =    Good.new(_('Copper'),    true, [  5, 8 ])
Mud =       Good.new(_('Mud'),       true, [  1, 5 ])
Tin =       Good.new(_('Tin'),       true, [  7, 9 ])
Gold =      Good.new(_('Gold'),      true, [ 15,16 ])
Clothes =   Good.new(_('Clothes'),   true, [ 12,13 ], [ Cotton ])
Beer =      Good.new(_('Beer'),      true, [ 10,11 ], [ Barley ])
Olive_oil = Good.new(_('Olive Oil'), true, [ 12,13 ], [ Olives ])
Furniture = Good.new(_('Furniture'), true, [ 10,11 ], [ Wood ])
Utensils =  Good.new(_('Utensils'),  true, [ 12,13 ], [ Copper ])
Bricks =    Good.new(_('Bricks'),    true, [  1, 8 ], [ Mud ])
Riding =    Good.new(_('Animals'),   true, [  5, 8 ])
Bronze =    Good.new(_('Bronze'),    true, [  9,10 ], [ Copper, Tin ])
Weapons =   Good.new(_('Weapons'),   true, [  6, 7 ], [ Wood, Bronze ])
Scrolls =   Good.new(_('Scrolls'),   false)
Prayers =   Good.new(_('Prayers'),   false)

# 
# Terrains
#
class Terrain

  @@all = []

  attr_reader :all, :name, :ovl, :cost_to_enter, :defensive_bonus, :production
  attr_accessor :switch_status

  def initialize(name, ovl, cost_to_enter, defensive_bonus, production, 
                 switch_status=nil)
    @name, @ovl, @cost_to_enter, @defensive_bonus, @switch_status =
      name, ovl, cost_to_enter, defensive_bonus, switch_status
    @production = {}
    i = 0
    Good.all.each do |good|
      @production[good] = production[i]
      i += 1
    end
    @switch_status.switch_status = self if @switch_status
    @@all << self
  end

end

#                      Name                    MoveCost       Barley      Mud
#                                           Ovl*   Defence       Olives      Tin
#                                                       Food        Wood        Gold
#                                                          Cotton      Copper
Tundra   = Terrain.new(_('Tundra'),          1, 0, 0, [ 2, 0, 0, 0, 0, 2, 2, 1, 0 ])
Desert   = Terrain.new(_('Desert'),         13, 1, 0, [ 1, 0, 0, 2, 0, 1, 3, 2, 0 ])
Plains   = Terrain.new(_('Planis'),         14, 1, 0, [ 4, 2, 1, 0, 0, 1, 1, 0, 0 ])
Prairie  = Terrain.new(_('Prairie'),        15, 1, 0, [ 3, 3, 0, 0, 0, 0, 0, 0, 0 ])
Steppe   = Terrain.new(_('Steppe'),         16, 1, 0, [ 2, 2, 3, 0, 0, 0, 0, 0, 0 ])
Marsh    = Terrain.new(_('Marsh'),          17, 2, 1, [ 3, 1, 2, 0, 0, 2, 4, 2, 0 ])

Boreal_f = Terrain.new(_('Boreal forest'),   6, 2, 2, [ 1, 0, 0, 0, 2, 1, 1, 1, 0 ], Tundra)
Scrub    = Terrain.new(_('Scrubland'),       7, 1, 2, [ 1, 0, 0, 4, 1, 1, 2, 1, 0 ], Desert)
Mixed_f  = Terrain.new(_('Mixed forest'),    8, 2, 2, [ 2, 1, 0, 0, 3, 0, 1, 0, 0 ], Plains)
Savannah = Terrain.new(_('Savannah'),        9, 1, 1, [ 1, 1, 0, 0, 2, 0, 1, 0, 0 ], Prairie)
Woodland = Terrain.new(_('Woodland'),       10, 2, 2, [ 1, 0, 1, 0, 2, 0, 1, 0, 0 ], Steppe)
Swamp    = Terrain.new(_('Swamp'),          11, 3, 3, [ 2, 0, 0, 0, 1, 1, 3, 2, 0 ], Marsh)

Arctic   = Terrain.new(_('Arctic'),          5, 2, 0, [ 0, 0, 0, 0, 0, 1, 0, 1, 1 ])
Sea      = Terrain.new(_('Sea'),             2, 1, 0, [ 3, 0, 0, 0, 0, 0, 0, 0, 0 ])
Ocean    = Terrain.new(_('Ocean'),           1, 1, 0, [ 3, 0, 0, 0, 0, 0, 0, 0, 0 ])
Hills    = Terrain.new(_('Hills'),           4, 2, 4, [ 1, 0, 0, 0, 0, 3, 0, 3, 0 ])
Mountain = Terrain.new(_('Mountains'),       3, 3, 6, [ 0, 0, 0, 0, 0, 3, 0, 3, 1 ])
# *Ovl: Overlap for terrains.


#
# Military classes
#
class Military

  @@all = []

  attr_reader :name, :ship, :moves, :attack, :defense, :cargo

  def initialize(name, ship, moves, attack, defense, cargo)
    @name, @ship, @moves, @attack, @defense, @cargo = name, ship, moves, attack, defense, cargo
    @@all << self
  end

end

Peasant         = Military.new(_('Peasant'),                 false, 1, 0, 1, 0)
Colonist        = Military.new(_('Colonist'),                false, 1, 0, 1, 0)
Warrior         = Military.new(_('Warrior'),                 false, 1, 2, 2, 0)
Nomad           = Military.new(_('Nomad'),                   false, 4, 1, 1, 0)
Chariot         = Military.new(_('Chariot'),                 false, 4, 3, 3, 0)
Pikeman         = Military.new(_('Pikeman'),                 false, 1, 3, 3, 0)
Mtd_pikeman     = Military.new(_('Mounted Pikeman'),         false, 4, 4, 4, 0)
Armored_warrior = Military.new(_('Armed Warrior'),           false, 1, 4, 4, 0)
Cataphract      = Military.new(_('Cataphract'),              false, 4, 5, 5, 0)
Catapult        = Military.new(_('Catapult'),                false, 1, 7, 5, 0)

Clan_member     = Military.new(_('Clan Member'),             false, 1, 1, 1, 0)
Clan_armed      = Military.new(_('Armed Clan Member'),       false, 1, 2, 2, 0)
Clan_mounted    = Military.new(_('Clan Member Rider'),       false, 4, 2, 2, 0)
Clan_mtd_armed  = Military.new(_('Armed Clan Member Armed'), false, 4, 3, 3, 0)

Elite_soldier   = Military.new(_('Elite Soldier'),           false, 1, 4, 4, 0)
Elite_chariot   = Military.new(_('Elite Chariot'),           false, 4, 5, 5, 0)
Faraoh_soldier  = Military.new(_("Faraoh's Guard"),          false, 1, 5, 5, 0)
Faraoh_chariot  = Military.new(_("Faraoh's Chariot Guard"),  false, 4, 6, 6, 0)

Penteconter     = Military.new(_('Penteconter'),             true,  4, 0, 2, 2)
Merchantman     = Military.new(_('Merchantman'),             true,  6, 0, 6, 5)
Monoreme        = Military.new(_('Monoreme'),                true,  4, 1, 3, 0)
Bireme          = Military.new(_('Bireme'),                  true,  5, 3, 5, 0)
Trireme         = Military.new(_('Trireme'),                 true,  6, 4, 6, 0)
Galley          = Military.new(_('Galley'),                  true,  5, 7, 8, 6)
Pirate_ship     = Military.new(_('Pirate Ship'),             true,  8, 4, 8, 2)


# 
# Jobs
#
class Job

  attr_reader :name
  attr_accessor :building, :raw_good, :good

  def initialize(name)
    @name = name
  end

end

Farmer       = Job.new(_('Farmer'))
Fisherman    = Job.new(_('Fisherman'))
Cotton_pl    = Job.new(_('Cotton Planter'))
Barley_pl    = Job.new(_('Barley Planter'))
Olive_pl     = Job.new(_('Olives Planter'))
Lumberjack   = Job.new(_('Lumberjack'))
Copper_miner = Job.new(_('Copper Miner'))
Mudder       = Job.new(_('Mudder'))
Tin_miner    = Job.new(_('Tin Miner'))
Prospector   = Job.new(_('Prospector'))
Weaver       = Job.new(_('Weaver'))
Brewer       = Job.new(_('Brewer'))
Oil_presser  = Job.new(_('Oil Presser'))
Carpenter    = Job.new(_('Carpenter'))
Coppersmith  = Job.new(_('Coppersmith'))
Potter       = Job.new(_('Potter'))
Metallurgic  = Job.new(_('Metallurgic'))
Weaponsmith  = Job.new(_('Weaponsmith'))
Scribe       = Job.new(_('Scribe'))
Priest       = Job.new(_('Priest'))

#
# Buildings
#
class Building

  attr_reader :name, :job, :raw_good, :good, :multiplier, :cost, :Copper, 
              :min_colony

  def initialize(name, job, raw_good, good, multiplier=1, cost=0, copper=0, 
                 min_colony=0)
    @name, @job, @raw_good, @good, @multiplier, @cost, @copper, @min_colony = 
      name, job, raw_good, good, multiplier, cost, copper, min_colony
    if @job
      @job.building = self
      @job.raw_good = raw_good
      @job.good = good
    end
  end

end

Weaver_1    = Building.new(_("Weaver's House"), Weaver, Cotton, Clothes)
Weaver_2    = Building.new(_("Weaver's Shop"), Weaver, Cotton, Clothes, 2, 64, 20)
Weaver_3    = Building.new(_("Cloth Mill"), Weaver, Cotton, Clothes, 3, 160, 100, 8)
Brewer_1    = Building.new(_("Brewer's House"), Brewer, Barley, Beer)
Brewer_2    = Building.new(_("Brewer's Shop"), Brewer, Barley, Beer, 2, 64, 20)
Brewer_3    = Building.new(_("Beer Mill"), Brewer, Barley, Beer, 3, 160, 100, 8)
Oil_1       = Building.new(_("Small Oil Press"), Oil_presser, Olives, Olive_oil)
Oil_2       = Building.new(_("Big Oil Press"), Oil_presser, Olives, Olive_oil, 2, 64, 20)
Oil_3       = Building.new(_("Olive Oil Mill"), Oil_presser, Olives, Olive_oil, 3, 160, 100, 8)
Carpenter_1 = Building.new(_("Carpenter's House"), Carpenter, Wood, Furniture)
Carpenter_2 = Building.new(_("Carpenter's Shop"), Carpenter, Wood, Furniture, 2, 64, 20)
Carpenter_3 = Building.new(_("Lumber Mill"), Carpenter, Wood, Furniture, 3, 160, 100, 8)
Copper_1    = Building.new(_("Coppersmith's House"), Coppersmith, Copper, Utensils)
Copper_2    = Building.new(_("Coppersmith's Shop"), Coppersmith, Copper, Utensils, 2, 64, 20)
Copper_3    = Building.new(_("Utensils Factory"), Coppersmith, Copper, Utensils, 3, 160, 100, 8)
Brick_1     = Building.new(_("Small Klin"), Potter, Mud, Bricks)
Brick_2     = Building.new(_("Big Klin"), Potter, Mud, Bricks, 2, 64, 20)
Brick_3     = Building.new(_("Brick Factory"), Potter, Mud, Bricks, 3, 160, 100, 8)
Stable      = Building.new(_("Stable"), nil, Food, Riding, 2, 64, 20)
Bronze_1    = Building.new(_("Metallurgic's Shop"), Metallurgic, [ Copper, Tin ], Bronze)
Bronze_2    = Building.new(_("Metalworks"), Metallurgic, [ Copper, Tin ], Bronze, 2, 64, 20)
Bronze_3    = Building.new(_("Machine Shop"), Metallurgic, [ Copper, Tin ], Bronze, 3, 160, 100, 8)
Scribe_1    = Building.new(_("Tablet House"), Scribe, nil, Scrolls)
Scribe_2    = Building.new(_("Gymnasium"), Scribe, nil, Scrolls, 2, 64, 20)
Scribe_3    = Building.new(_("Academy"), Scribe, nil, Scrolls, 3, 160, 100, 8)
Temple_1    = Building.new(_("Worship Yard"), Priest, nil, Prayers, 1, 80)
Temple_2    = Building.new(_("Shrine"), Priest, nil, Prayers, 2, 64, 20)
Temple_3    = Building.new(_("Temple"), Priest, nil, Prayers, 3, 160, 100, 8)
Embassy     = Building.new(_("Embassy"), nil, nil, nil, 3, 160, 100, 8)
Docks_1     = Building.new(_("Docks"), nil, nil, nil, 1, 52)
Docks_2     = Building.new(_("Drydock"), nil, nil, nil, 1, 80, 50, 4)
Docks_3     = Building.new(_("Shipyard"), nil, nil, nil, 1, 240, 100, 8)
Wall_1      = Building.new(_("Wooden Wall"), nil, nil, nil, 1, 64)
Wall_2      = Building.new(_("Stone Wall"), nil, nil, nil, 2, 120, 100)
Wall_3      = Building.new(_("Masonry Wall"), nil, nil, nil, 3, 240, 200)
Library     = Building.new(_("Library"), nil, nil, nil, 0, 120, 20)
Oracle      = Building.new(_("Oracle"), nil, nil, nil, 0, 120, 20)
Warehouse_1 = Building.new(_("Warehouse"), nil, nil, nil, 2, 80)
Warehouse_2 = Building.new(_("Big Warehouse"), nil, nil, nil, 3, 120, 20)
Palace      = Building.new(_("Palace"), nil, nil, nil, 0, 240, 200)
Monument    = Building.new(_("Monument"), nil, nil, nil, 0, 320, 300)
